libs/vkd3d/state.c

```diff
@@ -3458,6 +3519,69 @@ void vkd3d_fragment_output_pipeline_desc_prepare(struct vkd3d_fragment_output_pi
         desc->dy_info.pDynamicStates = desc->dy_states;
 }
 
+uint32_t vkd3d_fragment_output_pipeline_desc_hash(const void *key)
+{
+    return hash_data(key, sizeof(struct vkd3d_fragment_output_pipeline_desc));
+}
+
+bool vkd3d_fragment_output_pipeline_desc_compare(const void *key, const struct hash_map_entry *entry)
+{
+    const struct vkd3d_fragment_output_pipeline *pipeline = (const void*)entry;
+    const struct vkd3d_fragment_output_pipeline_desc *desc = key;
+    // We zero-initialize these so memcmp is fine
```

Comment:
C++ comment.

Reply:
Pointers are not set either, that might be worth commenting about as well. I was initially a bit confused why pointers were not set early but instead deferred to the "prepare" command.

Reply:
derp, fixed

Reply:
fwiw we do have a comment in `vkd3d_vertex_input_pipeline_desc_init` about pointers not being set.

Reply:
clarified comment